stubs/redis/redis/client.pyi

```diff
@@ -684,38 +684,297 @@ class PubSub:
     def load_scripts(self): ...
     def execute(self, raise_on_error: bool = ...) -> List[Any]: ...
-    def watch(self, *names: _Key) -> bool: ...
-    def unwatch(self): ...
     def script_load_for_pipeline(self, script): ...
-
-class StrictPipeline(BasePipeline, StrictRedis): ...
-class Pipeline(BasePipeline, Redis): ...
+    # methods from `Redis` client.
+    # arguments match `Redis`, but with some exceptions return types are
+    # `Pipeline` or `None`
+    def set_response_callback(self, command, callback): ...
+    def pipeline(self, transaction: bool = ..., shard_hint: Any = ...) -> Pipeline: ...
+    def lock(self, name, timeout=..., sleep=..., blocking_timeout=..., lock_class=..., thread_local=...): ...
+    def pubsub(self, shard_hint: Any = ..., ignore_subscribe_messages: bool = ...) -> PubSub: ...
+    def execute_command(self, *args, **options): ...
+    def parse_response(self, connection, command_name, **options): ...
+    def acl_cat(self, category: Optional[Text] = ...) -> Pipeline: ...
+    def acl_deluser(self, username: Text) -> Pipeline: ...
+    def acl_genpass(self) -> Pipeline: ...
+    def acl_getuser(self, username: Text) -> Pipeline: ...
+    def acl_list(self) -> Pipeline: ...
+    def acl_load(self) -> Pipeline: ...
+    def acl_setuser(
+        self,
+        username: Text = ...,
+        enabled: bool = ...,
+        nopass: bool = ...,
+        passwords: Optional[Sequence[Text]] = ...,
+        hashed_passwords: Optional[Sequence[Text]] = ...,
+        categories: Optional[Sequence[Text]] = ...,
+        commands: Optional[Sequence[Text]] = ...,
+        keys: Optional[Sequence[Text]] = ...,
+        reset: bool = ...,
+        reset_keys: bool = ...,
+        reset_passwords: bool = ...,
+    ) -> Pipeline: ...
+    def acl_users(self) -> Pipeline: ...
+    def acl_whoami(self) -> Pipeline: ...
+    def bgrewriteaof(self) -> Pipeline: ...
+    def bgsave(self) -> Pipeline: ...
+    def client_id(self) -> Pipeline: ...
+    def client_kill(self, address: Text) -> Pipeline: ...
+    def client_list(self) -> Pipeline: ...
+    def client_getname(self) -> Pipeline: ...
+    def client_setname(self, name: Text) -> Pipeline: ...
+    def readwrite(self) -> Pipeline: ...
+    def readonly(self) -> Pipeline: ...
+    def config_get(self, pattern=...) -> Pipeline: ...
+    def config_set(self, name, value) -> Pipeline: ...
+    def config_resetstat(self) -> Pipeline: ...
+    def config_rewrite(self) -> Pipeline: ...
+    def dbsize(self) -> Pipeline: ...
+    def debug_object(self, key) -> Pipeline: ...
+    def echo(self, value) -> Pipeline: ...
+    def flushall(self) -> Pipeline: ...
+    def flushdb(self) -> Pipeline: ...
+    def info(self, section: Optional[_Key] = ...) -> Pipeline: ...
+    def lastsave(self) -> Pipeline: ...
+    def object(self, infotype, key) -> Pipeline: ...
+    def ping(self) -> Pipeline: ...
+    def save(self) -> Pipeline: ...
+    def sentinel(self, *args) -> None: ...
```

Comment:
this deprecated method doesn't do or return anything: https://github.com/andymccurdy/redis-py/blob/1a41cfd95a53a95b078084d8627be6b6fba3bb71/redis/client.py#L1448-L1451